core/object/object.h

```diff
@@ -674,6 +680,7 @@ class Object {
 	virtual void _notificationv(int p_notification, bool p_reversed) {}
 
 	static void _bind_methods();
+	static void _bind_compatibility_methods();
```

Comment:
Perhaps it would be less awkward if you do something like...

```suggestion
#ifndef DISABLE_DEPRECATED
	static void _bind_compatibility_methods();
#else
	static void _bind_compatibility_methods() {}
#endif
```

Is that legal?